core/unix/rseq_linux.c

```diff
@@ -596,7 +596,11 @@ rseq_process_module(module_area_t *ma, bool at_map)
             goto rseq_process_module_cleanup;
         strtab = (char *)(str_map + sec_hdr[elf_hdr->e_shstrndx].sh_offset - offs);
     }
-    rseq_process_elf_sections(ma, at_map, sec_hdr, strtab, load_offs);
+    /* When saw_glibc_rseq_reg is set, we are still at glibc init, and ld has not
+     * relocated the executable yet.
```

Comment:
Seems like this should be a more general `exe_relocated`: or really the `at_map` should be set to true when dealing with the executable if it has not yet been relocated, as that is what `at_map`means: the module in question has been mmapped but not relocated and that is exactly what the kernel does.  So I think this extra bool param should be removed and instead `at_map` should be properly set.

Reply:
>  or really the at_map should be set to true when dealing with the executable if it has not yet been relocated

That causes a seg fault at https://github.com/DynamoRIO/dynamorio/blob/9fb14a72726fa1bf1c84812a10d6b3f7b69f0744/core/unix/rseq_linux.c#L475, probably because the section headers still need to be mapped in.

https://github.com/DynamoRIO/dynamorio/blob/9fb14a72726fa1bf1c84812a10d6b3f7b69f0744/core/unix/rseq_linux.c#L568

The comment above states that we delay rseq_process_module until we see the app use rseq to avoid such issues. But with glibc registering rseq so early, we don't have that option.

Reply:
About the name: it seemed better to not give a general name because seems that we're at a tricky state during libc init. So I gave it a very specific name `saw_glibc_rseq_reg` which seems sufficient for the  current case.